cmd/ctl/pkg/create/certificaterequest/certificaterequest.go

```diff
@@ -225,10 +255,44 @@ func (o *Options) Run(args []string) error {
 	}
 	req, err = o.CMClient.CertmanagerV1alpha2().CertificateRequests(ns).Create(context.TODO(), req, metav1.CreateOptions{})
 	if err != nil {
-		return fmt.Errorf("error creating CertificateRequest: %v", err)
+		return fmt.Errorf("error creating CertificateRequest: %w", err)
 	}
 	fmt.Fprintf(o.Out, "CertificateRequest %s has been created in namespace %s\n", req.Name, req.Namespace)
 
+	if o.FetchCert {
+		fmt.Fprintf(o.Out, "CertificateRequest %v in namespace %v has not been signed yet. Wait until it is signed...\n",
+			req.Name, req.Namespace)
+		err = wait.Poll(time.Second, o.Timeout, func() (done bool, err error) {
+			req, err = o.CMClient.CertmanagerV1alpha2().CertificateRequests(req.Namespace).Get(context.TODO(), req.Name, metav1.GetOptions{})
+			if err != nil {
+				return false, nil
+			}
+			return apiutil.CertificateRequestHasCondition(req, cmapiv1alpha2.CertificateRequestCondition{
+				Type:   cmapiv1alpha2.CertificateRequestConditionReady,
+				Status: cmmeta.ConditionTrue,
+			}), nil
+		})
+		if err != nil {
+			return fmt.Errorf("error when waiting for CertificateRequest to be signed: %w", err)
+		}
+		if len(req.Status.Certificate) == 0 {
+			fmt.Fprintf(o.Out, "%s\n", req.Status.Certificate)
```

Comment:
Printing `req.Status.Certificate` here doesn't seem useful as we know it is empty.